tests/plugins/api/test_plugin_hook_manager.py

```
@@ -106,3 +107,148 @@ def test_webhook_url_validation() -> None:
             event=HookEvent.FILE_SCANNED,
             callback_url="not-a-url",
         )
+
+    # Test SSRF block on localhost
+    with pytest.raises(ValueError, match="not allowed|Loopback"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)